### PR TITLE
Fix some object references

### PR DESCRIPTION
**Pull request by NikoCat233**
Not finished yet, dont merge

### CODE REVIEW

**Comment by Tommy-XL** [comment]
Wait

**Comment by Tommy-XL** [comment]
> Not finished yet, dont merge

So...

**Comment by Pietrodjaowjao** [comment]
dIDNT WORK

**Comment by Pietrodjaowjao** [comment]
KUNG FU PANDA

**Comment by Ultradragon005** [comment]
🤖

**Comment by Tommy-XL** [comment]
Still works for lasted version?

File: Modules/RPC.cs [hunk 1]
```diff
@@ -610,6 +611,131 @@ private static bool IsVersionMatch(int ClientId)
...
+[HarmonyPatch(typeof(PlayerPhysics), nameof(PlayerPhysics.HandleRpc))]
+internal class PlayerPhysicsRPCHandlerPatch
+{
+    private static bool hasVent(int ventId) => ShipStatus.Instance.AllVents.Any(v => v.Id == ventId);
+    private static bool hasLadder(int ladderId) => ShipStatus.Instance.Ladders.Any(l => l.Id == ladderId);
+
+    public static bool Prefix(PlayerPhysics __instance, byte callId, MessageReader reader)
+    {
+        var rpcType = (RpcCalls)callId;
+        MessageReader subReader = MessageReader.Get(reader);
```
Comment (Tommy-XL) [inline Modules/RPC.cs:623]: Better move EAC code to EAC.cs like this
image

**Pull request merged**